test/metatx/MinimalForwarder.test.js

```diff
@@ -124,43 +125,60 @@ contract('MinimalForwarder', function (accounts) {
       context('invalid signature', function () {
         it('tampered from', async function () {
           await expectRevert(
-            this.forwarder.execute({ ...this.req, from: accounts[0] }, this.sign),
+            this.forwarder.execute({ ...this.req, from: accounts[0] }, this.sign()),
             'MinimalForwarder: signature does not match request',
           );
         });
         it('tampered to', async function () {
           await expectRevert(
-            this.forwarder.execute({ ...this.req, to: accounts[0] }, this.sign),
+            this.forwarder.execute({ ...this.req, to: accounts[0] }, this.sign()),
             'MinimalForwarder: signature does not match request',
           );
         });
         it('tampered value', async function () {
           await expectRevert(
-            this.forwarder.execute({ ...this.req, value: web3.utils.toWei('1') }, this.sign),
+            this.forwarder.execute({ ...this.req, value: web3.utils.toWei('1') }, this.sign()),
             'MinimalForwarder: signature does not match request',
           );
         });
         it('tampered nonce', async function () {
           await expectRevert(
-            this.forwarder.execute({ ...this.req, nonce: this.req.nonce + 1 }, this.sign),
+            this.forwarder.execute({ ...this.req, nonce: this.req.nonce + 1 }, this.sign()),
             'MinimalForwarder: signature does not match request',
           );
         });
         it('tampered data', async function () {
           await expectRevert(
-            this.forwarder.execute({ ...this.req, data: '0x1742' }, this.sign),
+            this.forwarder.execute({ ...this.req, data: '0x1742' }, this.sign()),
             'MinimalForwarder: signature does not match request',
           );
         });
         it('tampered signature', async function () {
-          const tamperedsign = web3.utils.hexToBytes(this.sign);
+          const tamperedsign = web3.utils.hexToBytes(this.sign());
           tamperedsign[42] ^= 0xff;
           await expectRevert(
             this.forwarder.execute(this.req, web3.utils.bytesToHex(tamperedsign)),
             'MinimalForwarder: signature does not match request',
           );
         });
       });
+
+      it('bubble out of gas', async function () {
+        const receiver = await CallReceiverMock.new();
+        const gasAvailable = 100000;
+        this.req.to = receiver.address;
+        this.req.data = receiver.contract.methods.mockFunctionOutOfGas().encodeABI();
+        this.req.gas = 1000000;
+
+        await expectRevert.assertion(
+          this.forwarder.execute(this.req, this.sign(), { gas: gasAvailable }),
+        );
+
+        const { gasUsed } = await web3.eth.getBlock('latest')
+          .then(({ transactions }) => web3.eth.getTransactionReceipt(transactions.find(Boolean)));
```

Comment:
Let's please avoid `.then`:

```suggestion
        const { transactions } = await web3.eth.getBlock('latest');
        const { gasUsed } = await web3.eth.getTransactionReceipt(transactions.find(Boolean));
```

Additionally I think we should avoid `.find(Boolean)` and `.filter(Boolean)`. I don't think these expressions are intuitive. In this case, I don't understand what we're doing. Isn't there a single transaction in the block?